src/access/examples/ExampleAccessControlDelegate.cpp

```diff
@@ -0,0 +1,1101 @@
+                        }
+                    }
+                    else
+                    {
+                        *index = &storage - acl;
+                    }
+                }
+                return &storage;
+            }
+        }
+        return nullptr;
+    }
+
+    static EntryStorage * FindUsedInAcl(size_t index, const FabricIndex * fabricIndex)
+    {
+        index = FilterIndex(index, fabricIndex);
+        if (index < ArraySize(acl))
+        {
+            auto storage = acl + index;
+            if (storage->InUse())
+            {
+                return storage;
+            }
+        }
+        return nullptr;
+    }
+
+public:
+    // Pool support
+    static EntryStorage pool[kEntryStoragePoolSize];
+
+    static EntryStorage * Find(EntryStorage * candidate)
+    {
+        if (candidate && candidate->InPool())
+        {
+            return candidate;
+        }
+        for (auto & storage : pool)
+        {
+            if (!storage.InUse())
+            {
+                return &storage;
+            }
+        }
+        return nullptr;
+    }
+
+    bool InPool() const
+    {
+        constexpr auto end = pool + ArraySize(pool);
+        return pool <= this && this < end;
+    }
+
+public:
+    EntryStorage() = default;
+
+    void Init()
+    {
+        if (!mInUse)
+        {
+            Clear();
+            mInUse = true;
+        }
+    }
+
+    bool InUse() const { return mInUse; }
+
+    void Release()
+    {
+        if (InPool())
+        {
+            mInUse = false;
+        }
+    }
+
+    void Clear()
+    {
+        mInUse = false;
+        mFabricIndex = chip::kUndefinedFabricIndex;
+        mAuthMode = AuthMode::kPase;
+        mPrivilege = Privilege::kView;
+        memset(mSubjects, SubjectStorage::kEmptyByte, sizeof(mSubjects));
+#if defined(__GNUC__)
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wclass-memaccess"
+        memset(mTargets, TargetStorage::kEmptyByte, sizeof(mTargets));
+#pragma GCC diagnostic pop
+#else
+        for (auto & target : mTargets)
+        {
+            target.Clear();
+        }
+#endif
+    }
+
+private:
+    static size_t FilterIndex(size_t index, const FabricIndex * fabricIndex)
+    {
+        if (fabricIndex != nullptr)
+        {
+            size_t unfiltered = 0;
+            size_t filtered = 0;
+            for (auto & storage : acl)
+            {
+                if (storage.InUse())
+                {
+                    return ArraySize(acl);
+                }
+                if (storage.mFabricIndex == *fabricIndex)
+                {
+                    if (filtered == index)
+                    {
+                        break;
+                    }
+                    filtered++;
+                }
+                unfiltered++;
+            }
+            return unfiltered;
+        }
+        return index;
+    }
+
+public:
+    static constexpr int kMaxSubjects = 4;
+    static constexpr int kMaxTargets = 3;
```

Comment:
These should be configurable. They are also no max, but rather *min*. These are the *minimum* supported.

Reply:
I added configuration before running for covid test, missed this spot, fixed now. Here, it's the max that can be here. In the config, it could be renamed to min. (But in that case, it shouldn't be a config, as the min is per spec.) Arguably, I should keep the config as max (of the example code) and static_assert it against the spec min...